src/formatDate.js

```diff
@@ -42,15 +42,31 @@
  *   ['DD', 'MM', 'YYYY', '.'],
  * ) // '18.02.1997'
  *
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
+  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateCollector = {};
+  const newDateFormat = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateCollector[fromFormat[i]] = dateParts[i];
+  }
+
+  if (!fromFormat.includes("YY")) {
```

Comment:
The 'if' condition checks if 'fromFormat' does not include 'YY', but there is no equivalent check for 'YYYY'. This could lead to errors if 'fromFormat' includes 'YYYY' but not 'YY'.